tests/php/Core/KernelTest.php

```diff
@@ -81,4 +87,61 @@ public function testInvalidConfigDetection()
 
         $kernel->getConfigLoader()->getManifest();
     }
+
+    public function provideAllowedHostsWarning(): array
+    {
+        $scenarios = [
+            [
+                'config' => [],
+                'isCli' => true,
+                'shouldLog' => true,
+            ],
+            [
+                'config' => ['*'],
+                'isCli' => true,
+                'shouldLog' => false,
+            ],
+            [
+                'config' => ['www.example.com', 'example.org'],
+                'isCli' => true,
+                'shouldLog' => false,
+            ],
+        ];
+        // Test both in CLI and non-CLI context
+        foreach ($scenarios as $name => $scenario) {
+            $scenario['isCli'] = false;
+            $scenarios[$name . ' (non-CLI)'] = $scenario;
+        }
+        return $scenarios;
+    }
+
+    /**
+     * @dataProvider provideAllowedHostsWarning
+     */
+    public function testAllowedHostsWarning(array $config, bool $isCli, bool $shouldLog): void
+    {
+        // Prepare mock to check if a warning is logged or not
+        $mockLogger = $this->getMockBuilder(Logger::class)->setConstructorArgs(['testLogger'])->getMock();
+        $expectLog = $shouldLog ? $this->once() : $this->never();
+        // Throw an exception when a warning is logged so we can catch it
```

Comment:
Does this actually throw an exception ... somehow? I don't see where we catch it?

Reply:
Copy/paste error. Comment removed.